protocol-units/execution/opt-executor/src/executor/transaction_pipe.rs

```diff
@@ -43,55 +42,15 @@ impl Executor {
 		if let Some(request) = next {
 			match request {
 				MempoolClientRequest::SubmitTransaction(transaction, callback) => {
-					// Pre-execute Tx to validate its content.
-					// Re-create the validator for each Tx because it uses a frozen version of the ledger.
-					// let vm_validator = VMValidator::new(Arc::clone(&self.db.reader));
-					// let tx_result = vm_validator.validate_transaction(transaction.clone())?;
-
-					// let status = if let Some(vm_status) = tx_result.status() {
-					// 	// If the verification failed, return the error status.
-					// 	let ms = MempoolStatus::new(MempoolStatusCode::VmError);
-					// 	(ms, Some(vm_status))
-					// } else {
-					let status = {
-						// add to the mempool
-						{
-							let mut core_mempool = self.core_mempool.write().await;
-
-							debug!(
-								"Adding transaction to mempool: {:?} {:?}",
-								transaction,
-								transaction.sequence_number()
-							);
-							let status = core_mempool.add_txn(
-								transaction.clone(),
-								0,
-								transaction.sequence_number(),
-								TimelineState::NonQualified,
-								true,
-							);
-
-							match status.code {
-								MempoolStatusCode::Accepted => {
-									debug!("Transaction accepted: {:?}", transaction);
-								}
-								_ => {
-									debug!("Transaction not accepted: {:?}", status);
-									Err(TransactionPipeError::TransactionNotAccepted(status))?;
-								}
-							}
-						}
-
-						// send along to the receiver
-						transaction_channel
-							.send(transaction)
-							.await
-							.map_err(|e| anyhow::anyhow!("Error sending transaction: {:?}", e))?;
-
-						// report status
-						let ms = MempoolStatus::new(MempoolStatusCode::Accepted);
-						(ms, None)
-					};
+					let span = info_span!(
+						"submit_transaction",
+						seq = transaction.sequence_number(),
```

Comment:
Should have transaction hash.

Reply:
Also, let's use `sequence_number`.

Reply:
> Should have transaction hash.

I'm a bit grossed out by the API to get that out of `SignedTransaction`: you either call `.committed_hash()` directly in which case it gets also cached in the object, or you do `.clone().committed_hash()` like in other places in our code for some reason (to avoid the caching side-effect when logging statement is enabled vs. when it's not?). This looks a bit wasteful to do just for logging, but if there is no other way, I will put it there.